packages/server/graphql/mutations/helpers/notifySlack.ts

```diff
@@ -131,25 +203,25 @@ const upsertSlackMessage = async (
     const {channel} = convoInfo
     const {latest} = channel
     if (latest) {
-      const {ts, bot_profile} = latest
-      const {name} = bot_profile
-      if (name === 'Parabol') {
-        const timestamp = new Date(Number.parseFloat(ts) * 1000)
-        const ageThresh = new Date(Date.now() - ms('5m'))
-        if (timestamp >= ageThresh) {
-          // trigger update
-          const res = await manager.updateMessage(channelId, slackText, ts)
-          if (!res.ok) {
-            console.error(res.error)
-          }
-          return
+      const {ts} = latest
+      const timestamp = new Date(Number.parseFloat(ts) * 1000)
+      const ageThresh = new Date(Date.now() - ms('5m'))
+      if (timestamp >= ageThresh) {
+        // blocks need to be stringified when updating but not posting
+        const stringifiedBlocks = JSON.stringify(blocks)
+        const res = await manager.updateMessage(channelId, stringifiedBlocks, ts)
+        if (!res.ok) {
+          console.error(res.error)
+          const postRes = await manager.postMessage(channelId, blocks)
```

Comment:
I noticed that when changing the timer, a new notification is created rather than the previous one being edited. The reason is in `addSlackAuth`, we're using the Slack user id as the `channelId` so that we can send messages from the `@Parabol`. However, to be able to update a message, we need the DM `channelId`. 

Requesting user info on the bot or bot info doesn't return this so I've grabbed all the IMs and found the IM `channelId` with the Slackbot's user id (see [here](https://github.com/ParabolInc/parabol/blob/a78aa53259e1c303e5f0cd82ec393bf46d346219/packages/server/graphql/mutations/addSlackAuth.ts#L119)). 

To be able to read all the IM channels, I think we should need the `im:read` [scope](https://api.slack.com/scopes/im:read). While this works in development without the scope, I suspect this is the same bug as the private channels one where the dev & prod bots require different scopes.

I haven't adjusted the scopes in this PR. If the `im:read` scope is required, it will simply post new notifications when the timer is adjusted rather than update the old one.

Instead, I'll adjust the scope in [this PR](https://github.com/ParabolInc/parabol/pull/4964) so that we don't need to wait until the Slack app is approved to merge this PR.

Reply:
just outta curiosity, have you tried querying that one specific channel with something like conversations.info? it'd be absolutely amazing to avoid an extra scope if possible

Reply:
I responded to this in the PR thread but adding it here incase you didn't see it: 

Yup, I've tried [conversations.info](https://api.slack.com/methods/conversations.info) (just tested it again to be sure) but it requires a `channelId` which we don't have for the Slackbot. The incoming webhook response doesn't provide the Slackbot `channelId` and nor does [auth.test](https://api.slack.com/methods/auth.test), [bots.info](https://api.slack.com/methods/bots.info) or [users.info](https://api.slack.com/methods/users.info). It looks like we have to search through the IMs to get the Slackbot `channelId`. 

Alternatively, if we want to avoid requesting the `im:read` scope, we could scrap the update message logic. The update message logic happens when a user updates the timer and the default Slack channel is the `@Parabol` bot. If it's a default Slack channel is a group channel, we send a new message as we don't get the timestamp of the latest message. Sparing a user a notification from `@Parabol` is nice but the value add is quite small IMO